kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -491,7 +492,9 @@ protected void handleTopicMetadataRequest(KafkaHeaderAndRequest metadataHar,
 
         // Command response for all topics
         List<TopicMetadata> allTopicMetadata = Collections.synchronizedList(Lists.newArrayList());
-        List<Node> allNodes = Collections.synchronizedList(Lists.newArrayList());
+        Set<Node> allNodes = Collections.synchronizedSet(Sets.newHashSet());
```

Comment:
Why did you change `allNodes`'s type to `Set`? Is there any benefit? What I can think of is you want to remove repeated nodes. Please explain the reason.

Reply:
Here should keep the `List`, I have changed it back. Thanks